lib/adagrams.rb

```diff
@@ -0,0 +1,207 @@
+# frozen_string_literal: true
+
+LETTER_POOL = [
+  Array.new(1, 'A'),
+  Array.new(2, 'B'),
+  Array.new(2, 'C'),
+  Array.new(4, 'D'),
+  Array.new(12, 'E'),
+  Array.new(2, 'F'),
+  Array.new(3, 'G'),
+  Array.new(2, 'H'),
+  Array.new(9, 'I'),
+  Array.new(1, 'J'),
+  Array.new(1, 'K'),
+  Array.new(4, 'L'),
+  Array.new(2, 'M'),
+  Array.new(6, 'N'),
+  Array.new(8, 'O'),
+  Array.new(2, 'P'),
+  Array.new(1, 'Q'),
+  Array.new(6, 'R'),
+  Array.new(4, 'S'),
+  Array.new(6, 'T'),
+  Array.new(4, 'U'),
+  Array.new(2, 'V'),
+  Array.new(2, 'W'),
+  Array.new(1, 'X'),
+  Array.new(2, 'Y'),
+  Array.new(1, 'Z'),
+].feeze
+
+def pool_creation(letter_pool)
+  letter_pool_to_draw_from = []
+  letter_pool.each do |letters|
+    letter_pool_to_draw_from = letter_pool_to_draw_from.concat(letters)
+  end
+  letter_pool_to_draw_from
+end
+
+def draw_letters
+  letter_pool_to_draw_from = pool_creation(LETTER_POOL).shuffle
+  letters_in_hand = letter_pool_to_draw_from.pop(10)
+end
+
+def display_welcome_message
+  puts 'Welcome to Adagrams!'
+  puts "Let's draw 10 letters from the letter pool..."
+end
+
+def display_drawn_letters(letters)
+  puts 'You have drawn the letters:'
+  puts draw_letters.join(', ')
+end
+
+display_welcome_message
+display_drawn_letters(draw_letters)
+
+def uses_available_letters?(word, letters_in_hand)
+  letters_in_hand = letters_in_hand.clone
+  word = word.split('')
+  # Looping through word
+  word.each do |char|
+    # Remove char from hand
+    result = letters_in_hand.delete(char) # return nil if can't delete
+    return false if result == nil
+  end
+  return true
+end
+
+
+def score_word(word)
+  word = word.upcase
+  word = word.split('')
```

Comment:
You can use `chars` here to be more explicit:

```suggestion
  word = word.chars
```